sway/desktop/fx_renderer/fx_renderer.c

```diff
@@ -801,14 +820,14 @@ void fx_render_box_shadow(struct fx_renderer *renderer, const struct wlr_box *bo
 
 	glBlendFunc(GL_ONE, GL_ONE_MINUS_SRC_ALPHA);
 
-	glClearStencil(0);
-	glClear(GL_STENCIL_BUFFER_BIT);
-	glDisable(GL_STENCIL_TEST);
+	fx_renderer_stencil_mask_fini();
 }
 
 void fx_render_blur(struct fx_renderer *renderer, const float matrix[static 9],
 		struct fx_framebuffer **buffer, struct blur_shader *shader,
 		const struct wlr_box *box, int blur_radius) {
+	GLboolean initial_stencil;
+	glGetBooleanv(GL_STENCIL_TEST, &initial_stencil);
```

Comment:
We should probably avoid calling this 10+ times for every damage region:

https://www.khronos.org/opengl/wiki/Common_Mistakes#glGetFloatv_glGetBooleanv_glGetDoublev_glGetIntegerv